File: backend/app/schemas/analytics.py
"""Analytics API schemas."""

from pydantic import BaseModel


class AnalyticsOverview(BaseModel):
    """High-level LokSrijan platform statistics."""

    total_users: int
    total_challenges: int
    validated_challenges: int
    total_projects: int
    completed_projects: int
    total_institutions: int
    universities: int
    ngos: int
    industries: int
    total_beneficiaries: int
    average_impact_score: float

File: frontend/src/types/analytics.ts
/**
 * Analytics types — aligned with backend app/schemas/analytics.py
 */

export interface AnalyticsOverview {
  total_users: number;
  total_challenges: number;
  validated_challenges: number;
  total_projects: number;
  completed_projects: number;
  total_institutions: number;
  universities: number;
  ngos: number;
  industries: number;
  total_beneficiaries: number;
  average_impact_score: number;
}
